connectors/src/connectors/confluence/temporal/activities.ts

```diff
@@ -334,6 +337,58 @@ export async function confluenceUpsertPageActivity({
   await upsertConfluencePageInDb(connector.id, page, visitedAtMs);
 }
 
+export async function confluenceUpdatePagesParentIdsActivity(
+  connectorId: ModelId,
+  spaceId: string,
+  visitedAtMs: number
+) {
+  const connector = await Connector.findByPk(connectorId);
+  if (!connector) {
+    throw new Error(`Connector not found (id: ${connectorId})`);
+  }
+
+  const pages = await ConfluencePage.findAll({
+    attributes: ["id", "pageId", "parentId", "spaceId"],
+    where: {
+      connectorId,
+      spaceId,
+      lastVisitedAt: visitedAtMs,
+      parentId: {
+        [Op.not]: null,
+      },
+    },
+  });
+
+  logger.info(
+    {
+      connectorId,
+      confluencePagesCount: pages.length,
+    },
+    "Start updating pages parent ids."
+  );
+
+  for (const page of pages) {
+    // Retrieve parents using the internal ID, which aligns with the permissions
+    // view rendering and RAG requirements.
+    const parents = [makeConfluenceInternalPageId(page.pageId)];
+    // TODO:(2023-01-26 flav) We can cache the map used in `getConfluencePageParentIds`.
+    const parentIds = await getConfluencePageParentIds(connectorId, page);
```

Comment:
the quick fix for caching is to make the query for all pages here, and pass it to getconfluencepageparentsids